ui/src/table.test.tsx

```diff
@@ -208,13 +208,37 @@ describe('Table.tsx', () => {
   })
 
   describe('sort', () => {
+    it('Does not render sort arrow on sortable columns by default', () => {
+      const { container } = render(<XTable model={tableProps} />)
+
+      // the sortingIcon class is present even if sort icon is not rendered, therefore we check for its child presence
+      expect(container.querySelector('i[class*=sortingIcon]')!.firstChild).not.toBeTruthy()
+    })
+
+    it('Renders sort arrow after clicking on sortable column header', () => {
+      const { container } = render(<XTable model={tableProps} />)
+
+      fireEvent.click(container.querySelectorAll('.ms-DetailsHeader-cellTitle')[0])
+      expect(container.querySelector('i[class*=sortingIcon]')!.firstChild).toBeTruthy()
+    })
+
+    it('Checks if sort arrow disappears after clicking on another sortable column header', () => {
+      const { container } = render(<XTable model={tableProps} />)
+
+      fireEvent.click(container.querySelectorAll('.ms-DetailsHeader-cellTitle')[0])
+      expect(container.querySelector('i[class*=sortingIcon]')!.firstChild).toBeTruthy()
+
+      fireEvent.click(container.querySelectorAll('.ms-DetailsHeader-cellTitle')[1])
+      expect(container.querySelector('i[class*=sortingIcon]')!.firstChild).not.toBeTruthy()
+    })
+
     it('Sorts by column - string', () => {
       const { container, getAllByRole } = render(<XTable model={tableProps} />)
 
       expect(getAllByRole('gridcell')[0].textContent).toBe(cell11)
-      fireEvent.click(container.querySelector('i[class*=sortingIcon]')!)
+      fireEvent.click(container.querySelectorAll('.ms-DetailsHeader-cellTitle')[0])
```

Comment:
This should work as well.

```suggestion
      fireEvent.click(container.querySelector('.ms-DetailsHeader-cellTitle'))
```